pkg/crypto/signer_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package crypto_test
+
+import (
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+)
+
+func TestDefaultSigner(t *testing.T) {
+	testBytes := []byte("test string")
+	privKey, err := crypto.GenerateSecp256k1Key()
+	if err != nil {
+		t.Fatal()
+	}
+
+	signer := crypto.NewDefaultSigner(privKey)
+	signature, err := signer.Sign(testBytes)
+	if err != nil {
+		t.Fatal()
+	}
+
+	t.Run("sign & recover with valid signature", func(t *testing.T) {
+		pubKey, err := signer.Recover(signature, testBytes)
+		if err != nil {
+			t.Fatal()
+		}
+
+		if pubKey.X.Cmp(privKey.PublicKey.X) != 0 || pubKey.Y.Cmp(privKey.PublicKey.Y) != 0 {
+			t.Fatalf("wanted %v but got %v", pubKey, &privKey.PublicKey)
+		}
+	})
+
+	t.Run("sign & recover with invalid signature", func(t *testing.T) {
```

Comment:
Technically this tests that the public key returned from a valid signature is not equal to another random public key. There are also really invalid signatures (where there is no associated public key and `.Recover` would return an error) and the description makes it sound like this is being tested.

Reply:
Yup, I thought about it but decided not to test those cases, I can actually add them as well, or change the signature, we'll do one of those :)

Reply:
Fixed